cpp/src/arrow/array/data.h

```diff
@@ -23,6 +23,7 @@
 #include <memory>
 #include <utility>
 #include <vector>
+#include <arrow/util/range.h>
```

Comment:
nit: put this include with the rest of the Arrow includes (and use quotes to be consistent)

Reply:
done